### ISSUE TITLE
SelectControl should probably use onChange instead of onBlur

### ISSUE THREAD

**Issue by mkaz** [WordPress/gutenberg#2227]
When implementing SelectControl in Inspector Controls for the Gallery links, I noticed that using the onBlur method requires clicking out of the select being selected, even after a change is made in order to get the Update button to become clickable and save the change.

The onBlur should be switched to onChange which should resolve

**Comment by aduth** [comment]
Related:

- #2000
- #392
- https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md
- http://cita.disability.uiuc.edu/html-best-practices/auto/onchange.php
- http://www.themaninblue.com/writing/perspective/2004/10/19/

**Comment by afercia** [comment]
Selects behavior is different across operating systems. See related core Trac ticket:

Review the usage of the change event on select elements
https://core.trac.wordpress.org/ticket/40925

And see what happens on Windows when using `change` on a Select:
https://cloudup.com/iuFxQ7CkA7k

Ideally, all select elements should just use a button.

**Comment by youknowriad** [comment]
Added some notes in #2000  explaining why I prefer using onChange over onBlur in the SelectControl component. I'm closing this issue for now.

**Comment by afercia** [comment]
Worth noting all the a11y recommendations here to not use onChange apply only when the change triggers an unexpected change of context. For example, when only selecting an option from a select would change a view. That wouldn't be acceptable for a11y, for the reasons explained above and also in the jsx-a11y docs. However, I haven't find a similar case in Gutenberg so far.

If I'm missing something, and there are cases of unexpected changes, then the related select shouldn't use onChange.
